pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -417,14 +403,17 @@ func (nim *nodeInfoManager) updateNodeInfoObject(
 		}
 	}
 
-	// Clone driver list, omitting the driver that matches the given driverName,
-	// unless the driver is identical to information provided, in which case no update is necessary.
-	var newDriverInfos []csiv1alpha1.CSIDriverInfo
+	// Clone driver list, omitting the driver that matches the given driverName
+	newDriverInfos := []csiv1alpha1.CSIDriverInfo{}
 	for _, driverInfo := range nodeInfo.CSIDrivers {
-		if driverInfo.Driver == driverName {
-			prevTopologyKeys := sets.NewString(driverInfo.TopologyKeys...)
-			if driverInfo.NodeID == driverNodeID && prevTopologyKeys.Equal(topologyKeys) {
-				// No update needed
+		if driverInfo.Spec.Driver == driverName {
+			prevTopologyKeys := sets.NewString(driverInfo.Spec.TopologyKeys...)
+			if driverInfo.Spec.NodeID == driverNodeID &&
+				prevTopologyKeys.Equal(topologyKeys) &&
```

Comment:
nit: sets.NewString(driverInfo.Spec.TopologyKeys...).Equal(topologyKeys) and combine with the previous if statement.